### PR TITLE
MRG: Offset when decimating

### PR DESCRIPTION
**Pull request by Eric89GXL**
Taking over #2486. Ready for review/merge.

### CODE REVIEW

File: mne/epochs.py [hunk 1]
```diff
@@ -356,6 +356,12 @@ def decimate(self, decim, copy=False):
             The amount to decimate data.
         copy : bool
             If True, operate on and return a copy of the Epochs object.
+        offset : int
+            Apply an offset to where the decimation starts relative to the
+            sample corresponding to t=0. The offset is in samples at the
+            current sampling rate.
```
Comment (kingjr) [inline mne/epochs.py]: Maybe indicate that it is modula decim, or min = 0, max = decim -1

**Comment by kingjr** [comment]
LGTM

File: mne/epochs.py [hunk 2]
```diff
@@ -388,12 +394,13 @@ def decimate(self, decim, copy=False):
...
                  'frequency of %g Hz, which can cause aliasing artifacts.'
                  % (lowpass, decim, new_sfreq))  # > 50% nyquist lim
-
+        offset = int(offset) % decim
         epochs._decim *= decim
         start_idx = int(round(epochs._raw_times[0] * (epochs.info['sfreq'] *
                                                       epochs._decim)))
         i_start = start_idx % epochs._decim
-        decim_slice = slice(i_start, len(epochs._raw_times), epochs._decim)
+        decim_slice = slice(i_start + offset, len(epochs._raw_times),
```
Comment (agramfort) [inline mne/epochs.py]: not

len(epochs._raw_times) + offset ?

File: mne/epochs.py [hunk 3]
```diff
@@ -388,12 +394,13 @@ def decimate(self, decim, copy=False):
                  'of %g Hz. The decim=%i parameter will result in a sampling '
                  'frequency of %g Hz, which can cause aliasing artifacts.'
                  % (lowpass, decim, new_sfreq))  # > 50% nyquist lim
-
+        offset = int(offset) % decim
```
Comment (agramfort) [inline mne/epochs.py]: I would force user to have offset < decim. This is to me error prone.

File: mne/epochs.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/epochs.py]: no, you get at most `epochs._raw_times` out (indices should never go < 0 or >= len(epochs._raw_times))

**Comment by Eric89GXL** [comment]
@agramfort check added

**Comment by agramfort** [comment]
LGTM

merge when travis is green

thx !

**Pull request merged**